core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -474,6 +486,35 @@ private void tryStart(Instant startedAt) {
         }
     }
 
+    @VisibleForTesting
+    Checksum hashCopiedFiles() {
+        Checksum checksum = new Adler32();
+        copyToFileContainerPathMap.entrySet().stream().sorted(Entry.comparingByValue()).forEach(entry -> {
+            byte[] pathBytes = entry.getValue().getBytes();
+            // Add path to the hash
+            checksum.update(pathBytes, 0, pathBytes.length);
+
+            File file = new File(entry.getKey().getResolvedPath());
+            checksumFile(file, checksum);
+        });
+        return checksum;
+    }
+
+    @VisibleForTesting
+    @SneakyThrows(IOException.class)
+    void checksumFile(File file, Checksum checksum) {
+        checksum.update(MountableFile.getUnixFileMode(file.toPath()));
+        if (file.isDirectory()) {
+            try (Stream<Path> stream = Files.walk(file.toPath())) {
+                stream.filter(it -> !Files.isDirectory(it)).forEach(path -> {
```

Comment:
We have the file mode included in the checksum at line 506 which is cool...

But here, when we walk the contents, are we only walking sub_files_ (i.e. direct child files and child files of subdirectories)?

If that's the case then we'd not capture the file mode of subdirectories - is that the correct interpretation?

Reply:
Nice catch! Fixed.